guava/src/com/google/common/base/Absent.java

```diff
@@ -55,7 +58,9 @@ public T or(T defaultValue) {
   }
 
   @Override
-  public T or(Supplier<? extends T> supplier) {
+  @SuppressWarnings("nullness") //throws NullPointerException in case supplier returns null, which
```

Comment:
As per the error message, which says the following
>"use Optional.orNull() instead of a Supplier that returns null"

Is it safe to assume that supplier.get will always return a non-null value in this case?

Reply:
Are there any annotations similar to RequiresNonNull that can be used here to ensures that this method is being called only when suppplier.get is guaranteed to return a non-null value. What another possible approach can we employ to address this exception? Can it be assumed that 'or' method will only be called for suppliers that return non-null value?

Reply:
Only another way I could of think of is to restrict the return type to `@NonNull` for "supplier.get()" and we should not be doing that for, the return type should depend on the type `T`. Further using `@RequiresNonNull` will lead to `contracts.precondition.override.invalid` for overridden class Optional requires no such precondition for method "or". I guess we'll need to suppress the warning. Please correct me if I'm wrong